src/hooks/auth/useRequireAuth.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+
+import { useRouter } from 'next/router';
+
+import { isLoggedIn } from '@/utils/auth/login';
+
+/**
+ * Redirects to the login page if the user is not logged in.
+ */
+const useRequireAuth = () => {
```

Comment:
Amazing! can we use also apply this to `src/components/Collection/CollectionDetailContainer/CollectionDetailContainer.tsx` or remove the logic from `CollectionDetailContainer` and add it to: 

- `src/pages/collections/[collectionId]/index.tsx`
- `src/pages/collections/all/index.tsx`